restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -413,6 +423,8 @@ public void putAtPath(
       db.maybeCreateTableIndexes(keyspace, collection);
     }
 
+    checkTableIsDocsCollection(db, keyspace, collection);
```

Comment:
Sincere question: wouldn't this be checked in `DocumentResourceV2` as we already do with `keyspace` and `table`?

Reply:
Seems reasonable. Maybe all checks could be grouped into a single `Checker`. Maybe we can also cache the results there, I mean check should be done before doing any operation, but with each check we add extra latency. So if check is "green" for keyspace/table combo we could cache it for like 5-10mins? This way we would not recheck for tables that we know are fine.. And if something changes in the meantime, let's fail on the query then.. It's a tradeoff..

@EricBorczuk @eribeiro What do you both think?

Reply:
+1 to caching schema compatibility checks.